common/src/PCLPointCloud2.cpp

```diff
@@ -40,24 +40,25 @@
 
 #include <numeric>
 #include <vector>
+#include <cassert>
 
 #include <pcl/common/io.h>
 #include <pcl/pcl_macros.h>
 #include <pcl/exceptions.h>
 #include <pcl/PCLPointCloud2.h>
 
 bool
-pcl::PCLPointCloud2::concatenate (pcl::PCLPointCloud2 &cloud1, const pcl::PCLPointCloud2 &cloud2)
+pcl::PCLPointCloud2::concatenate (pcl::PCLPointCloud2 &cloud1, const pcl::PCLPointCloud2 &cloud2) noexcept
 {
   if (cloud1.is_bigendian != cloud2.is_bigendian)
   {
     // In future, it might be possible to convert based on pcl::getFieldSize(fields.datatype)
     PCL_ERROR ("[pcl::PCLPointCloud2::concatenate] Endianness of clouds does not match\n");
-    return (false);
+    return false;
   }
 
-  const auto size1 = cloud1.width * cloud1.height;
-  const auto size2 = cloud2.width * cloud2.height;
+  const std::size_t size1 = cloud1.width * cloud1.height;
```

Comment:
Maybe switch std::size_t to rhs to force the arithmetic in 64 bit mode

Reply:
Since we are here: what is the purpose of this check? I mean: is it enough checking that ``width * height`` of the two are the same? Or should we check that ``cloud1.width == cloud2.width && cloud1.height == cloud2.height`` ??

Reply:
I really don't know the reasoning behind a width and height when in several cases height is set to 1. It seems to be a legacy holdover. As a result, the strict check of equality is bad (`x.width == y.width && x.height == y.height`) and equivalence is better (`x.width*x.height == y.width*y.height`).

BTW, just to make sure my suggestion doesn't get lost in the discussion, I'm attaching a suggested diff
```suggestion
  const auto size1 = static_cast<std::size_t>(cloud1.width) * cloud1.height;
```

Reply:
> Since we are here: what is the purpose of this check? 

Sorry, don't understand which check you are referring to. But here is something to consider, maybe it will answer your question nevertheless. PCL supports two types of point clouds: organized and unorganized. The former has image-like structure, and the latter has `height` set to 1, so basically a flat vector of points. There is no meaningful way to concatenate two organized point clouds and preserve "organizedness", so the output will always be unorganized. And if the output in unorganized, then it does not really matter whether inputs were organized. And the right way to check whether two clouds (organized or not) have the same size is to compare their `width * height`s.